### ISSUE TITLE
`reportConstantRedefinition` false positive with branching logic

### ISSUE THREAD

**Issue by tylerlaprade** [microsoft/pyright#4979]
**Describe the bug**
If you conditionally initialize a constant inside an `if ... else` block, this rule thinks that the constant is redefined.

**To Reproduce** / **Screenshots or Code**
```python
if foo:
    MY_CONSTANT = 'a'
else:
    MY_CONSTANT = 'b'
```

**Expected behavior**
No warning
